Automattic/wp-calypso
# Framework: Move localStorage polyfill to separate bundle

As discovered in #7038 and #6558 before it, our local storage polyfill loading behavior is very fragile, and I suspect we're doomed to repeat it in its current state. As such, this pull request seeks to separate the polyfill into a separate bundle to be included in the page prior to the main application bundle.

**Testing instructions:**

Verify that your primary browser and Safari Private Mode load Calypso without any errors in the console.

**Open questions:**
- Are there desktop app considerations to be made here?
- Why does the polyfill bundle include vendor code? See related webpack/webpack#1016 and [examples/two-explicit-vendor-chunks](https://github.com/webpack/webpack/tree/master/examples/two-explicit-vendor-chunks) for reference
- **very important:** do we need systems to whitelist `public/polyfill.js` since it’s a static asset?

/cc @gwwar (fixed previous breakage) @blowery (tinkerer of Webpack)

Test live: https://calypso.live/?branch=add/polyfill-bundle
